ranges/typedranges.nim

```diff
@@ -132,10 +132,12 @@ proc `[]=`*[T, U, V](r: MutRange[T], s: HSlice[U, V], v: openarray[T]) =
   else:
     raise newException(RangeError, "different lengths for slice assignment")
 
+template toOpenArray*[T](r: Range[T]): auto =
+  # TODO: Casting through an {.unchecked.} array would be more appropriate
```

Comment:
The unchecked array might not work as expected on value types. I've found it works better on heap objects. With stack objects it requires the size of the item to be used even if you're using an unchecked array.